src/java.base/share/classes/java/io/BufferedOutputStream.java

```diff
@@ -89,15 +85,7 @@ private BufferedOutputStream(OutputStream out, int initialSize, int maxSize) {
             throw new IllegalArgumentException("Buffer size <= 0");
         }
 
-        if (getClass() == BufferedOutputStream.class) {
-            // use InternalLock and resizable buffer when not sub-classed
-            this.lock = InternalLock.newLockOrNull();
-            this.buf = new byte[initialSize];    // resizable
-        } else {
-            // use monitors and no resizing when sub-classed
-            this.lock = null;
-            this.buf = new byte[maxSize];
-        }
+        this.buf = new byte[initialSize]; // resizable if initialSize < maxSize
```

Comment:
Same remark as for BIS.

Also this initial vs max size feature was added when virtual thread was first implemented as the internal locks.  Is this implementation detail still necessary for the virtual threads?

Reply:
Think about 100_000 virtual threads, all blocked on sockets with input/output streams that have wrapped with buffered streams, as in BIS and BOS. This is the motivation for using a reduced initial size.

Reply:
```suggestion
```
Seems you forgot to remove this line after restoring the if-else below.